rlp/src/stream.rs

```diff
@@ -403,9 +413,9 @@ impl<'a> BasicEncoder<'a> {
 			// (prefix + length of length), followed by the length, followd by the string
 			len => {
 				self.buffer.put_u8(0);
-				let position = self.buffer.len();
+				let position = self.total_written();
 				let inserted_bytes = self.insert_size(len, position);
-				self.buffer[position - 1] = 0xb7 + inserted_bytes;
+				self.buffer[self.start_pos + position - 1] = 0xb7 + inserted_bytes;
```

Comment:
Same concern here, if both `start_pos` and `position` can be `0` (and `self.total_written()` starts at `0` right?) then we'd get an out of bounds panic right?

Reply:
see above